docs/examples/basic_meter/basic_metrics.py

```diff
@@ -50,13 +50,13 @@ def usage(argv):
 )
 
 
+# Stateful determines whether how metrics are collected: if true, metrics
+# accumulate over the process lifetime. If false, metrics are reset at the
+# beginning of each collection interval.
+metrics.set_meter_provider(MeterProvider(batcher_mode == "stateful"))
```

Comment:
Looks like this check happens on line 46. Right now this script crashes without any arguments

```suggestion
metrics.set_meter_provider(MeterProvider(stateful))
```

Reply:
This is a good question. The current design is such that the responsibility for "statefulness" belongs to the batcher, and in turn, the meter. This means that all metrics that created and recorded with this meter will have the same statefulness. This "batcher" concept was created previously when the metrics api was not as well defined (and we still don't really have a well defined SDK spec), in which we needed a way to determine whether we are calculating actual values or deltas. I think some of the [optional refinements](https://github.com/open-telemetry/oteps/blob/master/text/metrics/0088-metric-instrument-optional-refinements.md#aggregating-changes-to-a-sum-rate-calculation) for metric instruments is trying to address these cases, in which if they become actual api elements, we will have to change our design to decouple statefulness from the batcher and make them a metric level control.